src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -2687,15 +2685,51 @@ class VariantSampleMatrix(val hc: HailContext, val metadata: VSMMetadata,
       }
     }
 
-    (someIf(getVariants,
+    val irm = new IndexedRowMatrix(indexedRows, partStarts.last, nSamples)
```

Comment:
Unrelated, incidental changes are not free.  Try to balance the benefit (before and after here seem essentially the same to me) with the cost (it needs to be reviewed, there is danger to introduce bugs, increase changes of merge conflicts, etc.)

Reply:
Roger that, I've removed them.